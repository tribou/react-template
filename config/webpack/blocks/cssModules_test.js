// @flow

jest.mock('extract-text-webpack-plugin')

const webpack = require('webpack')
const config = require('./cssModules')

const mockContext = {
	fileType: () => {},
	platform: 'browser',
	webpack,
}
mockContext.fileType.add = () => {}

it('generates client dev config', () => {
	mockContext.platform = 'browser'
	const OLD_ENV = process.env.NODE_ENV
	process.env.NODE_ENV = 'development'

	expect(config()(mockContext)).toMatchSnapshot()
	process.env.NODE_ENV = OLD_ENV
})

it('generates client prod config', () => {
	mockContext.platform = 'browser'
	const OLD_ENV = process.env.NODE_ENV
	process.env.NODE_ENV = 'production'

	expect(config()(mockContext)).toMatchSnapshot()
	process.env.NODE_ENV = OLD_ENV
})

it('generates server dev config', () => {
	mockContext.platform = 'server'
	const OLD_ENV = process.env.NODE_ENV
	process.env.NODE_ENV = 'development'

	expect(config()(mockContext)).toMatchSnapshot()
	process.env.NODE_ENV = OLD_ENV
})

it('generates server prod config', () => {
	mockContext.platform = 'server'
	const OLD_ENV = process.env.NODE_ENV
	process.env.NODE_ENV = 'production'

	expect(config()(mockContext)).toMatchSnapshot()
	process.env.NODE_ENV = OLD_ENV
})
