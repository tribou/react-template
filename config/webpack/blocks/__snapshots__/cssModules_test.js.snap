// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates client dev config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": undefined,
        "use": Object {
          "extract": Object {
            "use": Array [
              Object {
                "loader": "css-loader",
                "options": Object {
                  "importLoaders": 1,
                  "localIdentName": "[path][name]__[local]__[hash:base64:3]",
                  "minimize": false,
                  "modules": true,
                },
              },
              Object {
                "loader": "postcss-loader",
                "options": Object {
                  "plugins": [Function],
                },
              },
            ],
          },
        },
      },
      Object {
        "exclude": undefined,
        "test": undefined,
        "use": Object {
          "extract": Object {
            "use": Array [
              Object {
                "loader": "css-loader",
                "options": Object {
                  "importLoaders": 1,
                  "localIdentName": "[path][name]__[local]__[hash:base64:3]",
                  "minimize": false,
                  "modules": false,
                },
              },
              Object {
                "loader": "postcss-loader",
                "options": Object {
                  "plugins": [Function],
                },
              },
            ],
          },
        },
      },
    ],
  },
  "plugins": Array [
    ExtractTextPlugin {
      "apply": [Function],
      "applyAdditionalInformation": [Function],
      "extract": [Function],
      "loader": [Function],
      "mergeNonInitialChunks": [Function],
      "renderExtractedChunk": [Function],
    },
  ],
}
`;

exports[`generates client prod config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": undefined,
        "use": Object {
          "extract": Object {
            "use": Array [
              Object {
                "loader": "css-loader",
                "options": Object {
                  "importLoaders": 1,
                  "localIdentName": "[hash:base64:8]",
                  "minimize": true,
                  "modules": true,
                },
              },
              Object {
                "loader": "postcss-loader",
                "options": Object {
                  "plugins": [Function],
                },
              },
            ],
          },
        },
      },
      Object {
        "exclude": undefined,
        "test": undefined,
        "use": Object {
          "extract": Object {
            "use": Array [
              Object {
                "loader": "css-loader",
                "options": Object {
                  "importLoaders": 1,
                  "localIdentName": "[hash:base64:8]",
                  "minimize": true,
                  "modules": false,
                },
              },
              Object {
                "loader": "postcss-loader",
                "options": Object {
                  "plugins": [Function],
                },
              },
            ],
          },
        },
      },
    ],
  },
  "plugins": Array [
    ExtractTextPlugin {
      "apply": [Function],
      "applyAdditionalInformation": [Function],
      "extract": [Function],
      "loader": [Function],
      "mergeNonInitialChunks": [Function],
      "renderExtractedChunk": [Function],
    },
  ],
}
`;

exports[`generates server dev config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": undefined,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "importLoaders": 0,
              "localIdentName": "[path][name]__[local]__[hash:base64:3]",
              "minimize": false,
              "modules": true,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
            },
          },
        ],
      },
      Object {
        "exclude": undefined,
        "test": undefined,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "importLoaders": 0,
              "localIdentName": "[path][name]__[local]__[hash:base64:3]",
              "minimize": false,
              "modules": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
            },
          },
        ],
      },
    ],
  },
}
`;

exports[`generates server prod config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "test": undefined,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "importLoaders": 0,
              "localIdentName": "[hash:base64:8]",
              "minimize": true,
              "modules": true,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
            },
          },
        ],
      },
      Object {
        "exclude": undefined,
        "test": undefined,
        "use": Array [
          Object {
            "loader": "css-loader/locals",
            "options": Object {
              "importLoaders": 0,
              "localIdentName": "[hash:base64:8]",
              "minimize": true,
              "modules": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
            },
          },
        ],
      },
    ],
  },
}
`;
